### PR TITLE
Safety net: add pytest

### PR DESCRIPTION
**Pull request by killthekitten**
As discussed in #15, I'm starting to build a safety net. I tried to keep the diff to minimum, but some formatters just don't want to collaborate.

I've fixed a couple bugs along the way, and found a potential one.

### CODE REVIEW

File: tests/test_coverage.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+from docstr_coverage import get_docstring_coverage
+
+SAMPLES_DIRECTORY = os.path.join("tests", "sample_files")
+EMPTY_FILE_PATH = os.path.join("tests", "sample_files", "empty_file.py")
+DOCUMENTED_FILE_PATH = os.path.join("tests", "sample_files", "documented_file.py")
+PARTLY_DOCUMENTED_FILE_PATH = os.path.join("tests", "sample_files", "partly_documented_file.py")
+SOME_CODE_NO_DOCS_FILE_PATH = os.path.join("tests", "sample_files", "some_code_no_docs.py")
+
+
+# @TODO: Should it check for module docs?
```
Comment (killthekitten) [inline tests/test_coverage.py:12]: @HunterMcGushion here's the potential bug I was referring to. Shouldn't it report a missing module doc in this case?

**Comment by HunterMcGushion** [comment]
Hey, @killthekitten, this looks awesome so far! Just to clarify, is this ready for review, or still a WIP?

**Comment by killthekitten** [comment]
@HunterMcGushion it's ready for a review. I didn't want to do everything in one PR.

**Comment by killthekitten** [comment]
@HunterMcGushion I guess there's a way to enable GH actions on your end - it doesn't seem to pick it up automatically.

File: tests/test_coverage.py (hunk 1 above)
Comment (HunterMcGushion) [reply tests/test_coverage.py:12]: To clarify, you're suggesting the log for the file should state both "File is empty" and "No module docstring"? I do understand that point, but in my opinion saying the file is empty already makes it clear that there's no module docstring. For the sake of being concise, [lines 233-241 in `coverage.py`](https://github.com/HunterMcGushion/docstr_coverage/blob/419eae9c5de7cfbdb38ff9c68705bdded1eb2f06/docstr_coverage/coverage.py#L233-L241) will log only one of {"File is empty", "No module docstring"}, or neither. Do you think logging both would be better?

**Comment by HunterMcGushion** [comment]
> @HunterMcGushion I guess there's a way to enable GH actions on your end - it doesn't seem to pick it up automatically.

According to the fourth paragraph in [the docs' "About GitHub Actions" section](https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions#about-github-actions), "Workflows in forked repositories don't run by default." So it seems like the workflow you added in this PR won't be recognized until it gets merged (I think?). 

A couple sections down, it says "By default, GitHub Actions is enabled on all repositories," and the repo settings confirm that Actions are enabled. So in my novice opinion, I think we'll need to wait for your PR to be merged to confirm Actions are working.

**Comment by killthekitten** [comment]
@HunterMcGushion I've [created a PR](https://github.com/killthekitten/docstr_coverage/pull/1) in my fork to check CI, all works now 👍

File: docstr_coverage/coverage.py [hunk 2]
```diff
@@ -306,7 +310,7 @@ def print_docstring(base, node, filename, ignore_names=()):
     log("Total docstring coverage: %.1f%%; " % (total_results["coverage"]), 1, True)
 
     #################### Calculate Total Grade ####################
-    grade = next(_[0] for _ in GRADES if _[1] <= total_results["coverage"])
+    grade = next((_[0] for _ in GRADES if _[1] <= total_results["coverage"]), GRADES[-1][0])
```
[the remark this answers is not in the document]
Comment (killthekitten) [reply docstr_coverage/coverage.py]: Thanks for pointing that out, it was a lazy fix.

This is for the case when the total coverage is 0 (empty files passed). With the original code, there's nothing to iterate over, so `next` raises a `StopIteration` error.

I've changed the code now, hope it makes more sense.

File: setup.py [hunk 3]
```diff
@@ -23,14 +23,12 @@ def readme():
     author="Hunter McGushion",
     author_email="hunter@mcgushion.com",
     license="MIT",
-    packages=["docstr_coverage"],
+    packages=find_packages(exclude=("tests",)),
```
[the remark this answers is not in the document]
Comment (killthekitten) [reply setup.py]: Sure, thanks for the context. I'll follow up in #1

File: tests/test_coverage.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
+import os
+
+from docstr_coverage import get_docstring_coverage
+
+SAMPLES_DIRECTORY = os.path.join("tests", "sample_files")
```
[the remark this answers is not in the document]
Comment (killthekitten) [reply tests/test_coverage.py:5]: Good catch! Actually, it was intended to test passing a directory to `get_docstring_coverage`. But then I realised directory traversal is happening in [`_execute`](https://github.com/HunterMcGushion/docstr_coverage/blob/master/docstr_coverage/coverage.py#L413).

File: tests/test_coverage.py (hunk 1 above)
Comment (killthekitten) [reply tests/test_coverage.py:12]: Sorry for not making it clear, and thanks for clarifying. The messaging is good as is.

I haven't realised that there is a special case of `File is empty`, which makes perfect sense. It's not reflected in the returned result dicts, that's what confused me.

**Comment by killthekitten** [comment]
@HunterMcGushion I've updated the PR with some additional commits. Let me know if it's good to go, I'll squash some of the commits before it's merged.

File: setup.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (killthekitten) [reply setup.py]: Followed up. I guess `find_packages` in this case is not critical. I.e. the requests library [does not use it](https://github.com/psf/requests/blob/master/setup.py#L42).

**Comment by HunterMcGushion** [comment]
> @HunterMcGushion I've [created a PR](https://github.com/killthekitten/docstr_coverage/pull/1) in my fork to check CI, all works now 👍

Ahh well played, sir!

File: tests/test_coverage.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (HunterMcGushion) [reply tests/test_coverage.py:5]: Hmmm I suppose that is an awkward place to put the directory traversal. I've got some refactoring to do.

File: docstr_coverage/coverage.py [hunk 5]
```diff
@@ -306,7 +310,11 @@ def print_docstring(base, node, filename, ignore_names=()):
     log("Total docstring coverage: %.1f%%; " % (total_results["coverage"]), 1, True)
 
     #################### Calculate Total Grade ####################
-    grade = next(_[0] for _ in GRADES if _[1] <= total_results["coverage"])
+    grade = next(
+        message
+        for message, grade_threshold in GRADES
+        if total_results["coverage"] <= grade_threshold
```
Comment (HunterMcGushion) [inline docstr_coverage/coverage.py:316]: Looks like these two got switched. Nice clarification, though!
```suggestion
        if grade_threshold <= total_results["coverage"]
```

Comment (killthekitten) [reply docstr_coverage/coverage.py:316]: Ah no, actually this was the source of the bug, that's why I switched these. This way the minimal grade_threshold would be `1`, while `total_results["coverage"]` can become `0`.

Comment (HunterMcGushion) [reply docstr_coverage/coverage.py:316]: Ahh I see. What if we make the suggested change, then also change the last entry in `GRADES` to be `("Do you even docstring?", 0),`. This should take care of the bug you mentioned while also keeping the message accurate. In its current state, the "Amazing" message always gets printed.

Comment (killthekitten) [reply docstr_coverage/coverage.py]: what a wonderful safety net I've built 🤦 thanks!

**Pull request merged**